moveit_core/CMakeLists.txt

```diff
@@ -175,16 +175,16 @@ catkin_package(
 include_directories(SYSTEM ${EIGEN3_INCLUDE_DIRS}
                            ${LIBFCL_INCLUDE_DIRS}
                            ${BULLET_INCLUDE_DIRS}
-                           )
+                           ${Boost_INCLUDE_DIRS}
+                           ${catkin_INCLUDE_DIRS}
+                           ${urdfdom_INCLUDE_DIRS}
+                           ${urdfdom_headers_INCLUDE_DIRS}
+                           ${OCTOMAP_INCLUDE_DIRS}
+                    )
```

Comment:
Including `catkin_INCLUDE_DIRS` as SYSTEM feels wrong. This also includes packages in our domain.
Why did you need to do that?

If you insist on making all of them SYSTEM includes, please sort them the way they originally where (previous SYSTEM includes should be appended at the end of the list instead of the beginning).

Reply:
The new gcc compiler introduces a whole bunch of new warnings. Many of them are related to external includes, which we cannot fix. Hence, in order to silent them, we need to declare those includes as external.
My changes keep the includes of each package as local includes.

Where did you notice a reordering of include folders? I thought I took care of placing `catkin_INCLUDE_DIRS` first.

Reply:
Looks like, I considered the order only later in my editing process. Will fix the remaining as well.

Reply:
Fixed.